qiskit_dynamics/solver_classes.py

```diff
@@ -0,0 +1,335 @@
+            raise QiskitError(
+                """Simulating SuperOp for a LinbladModel requires setting
+                vectorized evaluation. Set evaluation_mode to a vectorized option.
+                """
+            )
+
+        # modify initial state for some custom handling of certain scenarios
+        y_input = y0
+
+        # if Simulating density matrix or SuperOp with a HamiltonianModel, simulate the unitary
+        if y0_cls in [DensityMatrix, SuperOp] and isinstance(self.model, HamiltonianModel):
+            y0 = np.eye(self.model.dim, dtype=complex)
+        # if LindbladModel is vectorized and simulating a density matrix, flatten
+        elif (
+            (y0_cls is DensityMatrix)
+            and isinstance(self.model, LindbladModel)
+            and "vectorized" in self.evaluation_mode
+        ):
+            y0 = y0.flatten(order="F")
+
+        # validate y0 shape before passing to solve_lmde
+        if isinstance(self.model, HamiltonianModel):
+            if y0.shape[0] != self.model.dim or y0.ndim > 2:
+                raise QiskitError("""Shape mismatch for initial state y0 and HamiltonianModel.""")
+        if isinstance(self.model, LindbladModel):
+            if ("vectorized" not in self.model.evaluation_mode) and (
+                y0.shape[-2:] != (self.model.dim, self.model.dim)
+            ):
+                raise QiskitError("""Shape mismatch for initial state y0 and LindbladModel.""")
+            if ("vectorized" in self.model.evaluation_mode) and (
+                y0.shape[0] != self.model.dim ** 2 or y0.ndim > 2
+            ):
+                raise QiskitError(
+                    """Shape mismatch for initial state y0 and LindbladModel
+                                     in vectorized evaluation mode."""
+                )
+
+        results = solve_lmde(generator=self.model, t_span=t_span, y0=y0, **kwargs)
+
+        # handle special cases
+        if y0_cls is DensityMatrix and isinstance(self.model, HamiltonianModel):
+            # conjugate by unitary
+            out = Array(results.y)
+            results.y = out @ y_input @ out.conj().transpose((0, 2, 1))
+        elif y0_cls is SuperOp and isinstance(self.model, HamiltonianModel):
+            # convert to SuperOp and compose
+            out = Array(results.y)
+            results.y = (
+                np.einsum("nka,nlb->nklab", out.conj(), out).reshape(
+                    out.shape[0], out.shape[1] ** 2, out.shape[1] ** 2
+                )
+                @ y_input
+            )
+        elif (
+            (y0_cls is DensityMatrix)
+            and isinstance(self.model, LindbladModel)
+            and "vectorized" in self.evaluation_mode
+        ):
+            results.y = Array(results.y).reshape((len(results.y),) + y_input.shape, order="F")
+
+        if y0_cls is not None:
+            results.y = [final_state_converter(yi, y0_cls) for yi in results.y]
+
+        return results
+
+
+def initial_state_converter(
```

Comment:
If this isn't used anywhere else it could be a static class method:
```python
@staticmethod
def _initial_state_converter(obj)
```

Reply:
I think these methods will be useful in other places when we add other solvers, so not sure if we want them to appear specifically as a part of this class.